![board game](board_game.jpg)

# Which board game should you play?

## 📖 Background

After a tiring week, what better way to unwind than a board game night with friends and family? But the question remains: which game should you pick? You have gathered a dataset containing information of over `20,000` board games. It's time to utilize your analytical skills and use data-driven insights to persuade your group to try the game you've chosen!

## 💾 The Data

You've come across a dataset titled `bgg_data.csv` containing details on over `20,000` ranked board games from the BoardGameGeek (BGG) website. BGG is the premier online hub for board game enthusiasts, hosting data on more than `100,000` games, inclusive of both ranked and unranked varieties. This platform thrives due to its active community, who contribute by posting reviews, ratings, images, videos, session reports, and participating in live discussions.

This specific dataset, assembled in `February 2021`, encompasses all ranked games listed on BGG up to that date. Games without a ranking were left out because they didn't garner enough reviews; for a game to earn a rank, it needs a minimum of `30` votes.

In this dataset, each row denotes a board game and is associated with some information.

| Column     | Description              |
|------------|--------------------------|
| `ID` | The ID of the board game. |
| `Name` | The name of the board game.|
| `Year Published` | The year when the game was published.|
| `Min Players` | The minimum number of player recommended for the game.|
| `Max Players` | The maximum number of player recommended for the game.|
| `Play Time` | The average play time suggested by game creators, measured in minutes.|
| `Min Age` | The recommended minimum age of players.|
| `Users Rated` | The number of users who rated the game.|
| `Rating Average` | The average rating of the game, on a scale of 1 to 10.|
| `BGG Rank` | The rank of the game on the BoardGameGeek (BGG) website.| 
| `Complexity Average` | The average complexity value of the game, on a scale of 1 to 5.|
| `Owned Users` |  The number of BGG registered owners of the game.| 
| `Mechanics` | The mechanics used by the game.| 
| `Domains` | The board game domains that the game belongs to.|

**Source:** Dilini Samarasinghe, July 5, 2021, "BoardGameGeek Dataset on Board Games", IEEE Dataport, doi: https://dx.doi.org/10.21227/9g61-bs59.

# Executive Summary:

The top ten recommended boardgames are:...

In [2]:
import pandas as pd
boardgame = pd.read_csv('data/bgg_data.csv')
boardgame


ID                               Name  Year Published  \
0      174430.0                         Gloomhaven          2017.0   
1      161936.0          Pandemic Legacy: Season 1          2015.0   
2      224517.0                  Brass: Birmingham          2018.0   
3      167791.0                  Terraforming Mars          2016.0   
4      233078.0  Twilight Imperium: Fourth Edition          2017.0   
...         ...                                ...             ...   
20338   16398.0                                War             0.0   
20339    7316.0                              Bingo          1530.0   
20340    5048.0                         Candy Land          1949.0   
20341    5432.0                 Chutes and Ladders          -200.0   
20342   11901.0                        Tic-Tac-Toe         -1300.0   

       Min Players  Max Players  Play Time  Min Age  Users Rated  \
0                1            4        120       14        42055   
1                2            4         60       13        41643   
2                2            4        120       14        19217   
3                1            5        120       12        64864   
4                3            6        480       14        13468   
...            ...          ...        ...      ...          ...   
20338            2            2         30        4         1340   
20339            2           99         60        5         2154   
20340            2            4         30        3         4006   
20341            2            6         30        3         3783   
20342            2            2          1        4         3275   

       Rating Average  BGG Rank  Complexity Average  Owned Users  \
0                8.79         1                3.86      68323.0   
1                8.61         2                2.84      65294.0   
2                8.66         3                3.91      28785.0   
3                8.43         4                3.24      87099.0   
4                8.70         5                4.22      16831.0   
...               ...       ...                 ...          ...   
20338            2.28     20340                0.01        427.0   
20339            2.85     20341                1.05       1533.0   
20340            3.18     20342                1.08       5788.0   
20341            2.86     20343                1.02       4400.0   
20342            2.68     20344                1.16       1374.0   

                                               Mechanics  \
0      Action Queue, Action Retrieval, Campaign / Bat...   
1      Action Points, Cooperative Game, Hand Manageme...   
2      Hand Management, Income, Loans, Market, Networ...   
3      Card Drafting, Drafting, End Game Bonuses, Han...   
4      Action Drafting, Area Majority / Influence, Ar...   
...                                                  ...   
20338                                                NaN   
20339   Betting and Bluffing, Bingo, Pattern Recognition   
20340                               Roll / Spin and Move   
20341  Dice Rolling, Grid Movement, Race, Roll / Spin...   
20342                 Paper-and-Pencil, Pattern Building   

                                Domains  
0        Strategy Games, Thematic Games  
1        Strategy Games, Thematic Games  
2                        Strategy Games  
3                        Strategy Games  
4        Strategy Games, Thematic Games  
...                                 ...  
20338                  Children's Games  
20339                       Party Games  
20340                  Children's Games  
20341                  Children's Games  
20342  Abstract Games, Children's Games  

[20343 rows x 14 columns]

In [3]:
df_bg = boardgame.copy()
df_bg.head(20)


ID                                           Name  Year Published  \
0   174430.0                                     Gloomhaven          2017.0   
1   161936.0                      Pandemic Legacy: Season 1          2015.0   
2   224517.0                              Brass: Birmingham          2018.0   
3   167791.0                              Terraforming Mars          2016.0   
4   233078.0              Twilight Imperium: Fourth Edition          2017.0   
5   291457.0                   Gloomhaven: Jaws of the Lion          2020.0   
6   182028.0  Through the Ages: A New Story of Civilization          2015.0   
7   220308.0                                   Gaia Project          2017.0   
8   187645.0                           Star Wars: Rebellion          2016.0   
9    12333.0                              Twilight Struggle          2005.0   
10  193738.0                            Great Western Trail          2016.0   
11  115746.0                War of the Ring: Second Edition          2012.0   
12  162886.0                                  Spirit Island          2017.0   
13  169786.0                                         Scythe          2016.0   
14   84876.0                        The Castles of Burgundy          2011.0   
15  120677.0                                  Terra Mystica          2012.0   
16  173346.0                                 7 Wonders Duel          2015.0   
17  124361.0                                      Concordia          2013.0   
18   28720.0                              Brass: Lancashire          2007.0   
19  266192.0                                       Wingspan          2019.0   

    Min Players  Max Players  Play Time  Min Age  Users Rated  Rating Average  \
0             1            4        120       14        42055            8.79   
1             2            4         60       13        41643            8.61   
2             2            4        120       14        19217            8.66   
3             1            5        120       12        64864            8.43   
4             3            6        480       14        13468            8.70   
5             1            4        120       14         8392            8.87   
6             2            4        120       14        23061            8.43   
7             1            4        150       12        16352            8.49   
8             2            4        240       14        23081            8.42   
9             2            2        180       13        40814            8.29   
10            2            4        150       12        29457            8.30   
11            2            4        180       13        13725            8.49   
12            1            4        120       13        25722            8.35   
13            1            5        115       14        57871            8.24   
14            2            4         90       12        46507            8.13   
15            2            5        150       12        39284            8.14   
16            2            2         30       10        60302            8.11   
17            2            5        100       13        28105            8.12   
18            2            4        120       14        19400            8.17   
19            1            5         70       10        43165            8.11   

    BGG Rank  Complexity Average  Owned Users  \
0          1                3.86      68323.0   
1          2                2.84      65294.0   
2          3                3.91      28785.0   
3          4                3.24      87099.0   
4          5                4.22      16831.0   
5          6                3.55      21609.0   
6          7                4.41      26985.0   
7          8                4.35      20312.0   
8          9                3.71      34849.0   
9         10                3.59      56219.0   
10        11                3.71      35804.0   
11        12                4.14      22281.0   
12        1

Save a copy of the columns in a list for easier sub-grouping;

In [24]:
cols = df_bg.columns.to_list()
cols


['ID',
 'Name',
 'Year Published',
 'Min Players',
 'Max Players',
 'Play Time',
 'Min Age',
 'Users Rated',
 'Rating Average',
 'BGG Rank',
 'Complexity Average',
 'Owned Users',
 'Mechanics',
 'Domains']

In [37]:
df_bg.shape


(20343, 14)

Wonderful! As promised, we have just over 20,000 boardgames listed in the dataset.  Now let's check the status of the data with `.info()`

In [11]:
df_bg.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20343 entries, 0 to 20342
Data columns (total 14 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   ID                  20327 non-null  float64
 1   Name                20343 non-null  object 
 2   Year Published      20342 non-null  float64
 3   Min Players         20343 non-null  int64  
 4   Max Players         20343 non-null  int64  
 5   Play Time           20343 non-null  int64  
 6   Min Age             20343 non-null  int64  
 7   Users Rated         20343 non-null  int64  
 8   Rating Average      20343 non-null  float64
 9   BGG Rank            20343 non-null  int64  
 10  Complexity Average  20343 non-null  float64
 11  Owned Users         20320 non-null  float64
 12  Mechanics           18745 non-null  object 
 13  Domains             10184 non-null  object 
dtypes: float64(5), int64(6), object(3)
memory usage: 2.2+ MB


In [38]:
df_bg.isnull().sum()


ID                       16
Name                      0
Year Published            1
Min Players               0
Max Players               0
Play Time                 0
Min Age                   0
Users Rated               0
Rating Average            0
BGG Rank                  0
Complexity Average        0
Owned Users              23
Mechanics              1598
Domains               10159
dtype: int64

Okay interesting;

Above we can see that the total number of entries numbered 20343 is echoed in the same number of game titles without any null values.  There are however, 16 `null` values in the ID numbers.  

In [16]:
df_bg[df_bg['ID'].isnull()]


ID                                               Name  Year Published  \
10776 NaN                            Ace of Aces: Jet Eagles          1990.0   
10835 NaN                                 Die Erben von Hoax          1999.0   
11152 NaN  Rommel in North Africa: The War in the Desert ...          1986.0   
11669 NaN                  Migration: A Story of Generations          2012.0   
12649 NaN                   Die Insel der steinernen Wachter          2009.0   
12764 NaN                   Dragon Ball Z TCG (2014 edition)          2014.0   
13282 NaN                                           Dwarfest          2014.0   
13984 NaN                                                Hus             NaN   
14053 NaN                                        Contrario 2          2006.0   
14663 NaN                           Warage: Extended Edition          2017.0   
16292 NaN                                            Rainbow          2010.0   
17009 NaN                Sexy, el juego del arte del flirteo          2014.0   
18672 NaN                    Dracarys Dice Don't Get Burned!          2018.0   
19332 NaN  Battleship: Tactical Capital Ship Combat 1925-...          1986.0   
19474 NaN                          The Umbrella Academy Game          2020.0   
20040 NaN                                    Hidden Conflict          2005.0   

       Min Players  Max Players  Play Time  Min Age  Users Rated  \
10776            2            2         20       10          110   
10835            3            8         45       12          137   
11152            2            2          0       12           53   
11669            2            4         30       12           49   
12649            2            4        120       12           49   
12764            2            2         20        8           33   
13282            2            6         45       12           82   
13984            2            2         40        0           38   
14053            2           12          0       14           37   
14663            2            6         90       10           49   
16292            2            6         20        8           35   
17009            3            4        120       18           39   
18672            2            5         15       10           86   
19332            2            2         90        8          131   
19474            1            6         40       12           41   
20040            2            6         30       10          128   

       Rating Average  BGG Rank  Complexity Average  Owned Users Mechanics  \
10776            6.26     10778                0.02          NaN       NaN   
10835            6.05     10837                0.02          NaN       NaN   
11152            6.76     11154                0.04          NaN       NaN   
11669            7.20     11671                2.00          NaN       NaN   
12649            6.73     12651                0.03          NaN       NaN   
12764            7.03     12766                2.50          NaN       NaN   
13282            6.13     13284                1.75          NaN       NaN   
13984            6.28     13986                0.02          NaN       NaN   
14053            6.30     14055                1.00          NaN       NaN   
14663            7.64     14665                0.03          NaN       NaN   
16292            5.65     16294                0.02          NaN       NaN   
17009            5.83     17011                0.02          NaN       NaN   
18672            5.33     18674                1.00          NaN       NaN   
19332            5.07     19334                0.02          NaN       NaN   
19474            3.16     19476                1.00          NaN       NaN   
20040            4.35     20042                0.02          NaN       NaN   

      Domains  
10776     NaN  
10835     NaN  
11152     NaN  
11669     NaN  
12649     NaN  
12764     NaN  
13282     NaN  
13984     NaN  
14053     NaN  
14

### How many of the titles and ID numbers are unique?

In [9]:
df_bg.nunique()


ID                    20327
Name                  19976
Year Published          188
Min Players              11
Max Players              54
Play Time               116
Min Age                  21
Users Rated            2973
Rating Average          627
BGG Rank              20343
Complexity Average      383
Owned Users            3997
Mechanics              7381
Domains                  39
dtype: int64

### Do the duplicated game names also have duplicate data such as ID etc, or are they truly different games with the same name?

In [51]:
df_bg[['ID', 'Name']].value_counts().head(10)


ID        Name                                       
1.0       Die Macher                                     1
165041.0  Cargotrain                                     1
165190.0  Boom Bokken                                    1
165189.0  Altaria: Clash of Dimensions                   1
165186.0  Hitler's Reich: WW2 in Europe                  1
165095.0  Pirate Loot: Base Set                          1
165090.0  CLUE: Firefly Edition                          1
165046.0  Slavika: Equinox                               1
165044.0  EverZone: Strategic Battles in the Universe    1
165022.0  €uro Crisis                                    1
dtype: int64

Doesn't look like it, they're probably unique games then.  Lets take a closer look at an example:

In [52]:
df_bg['Name'].value_counts().head(10)


Robin Hood          6
Gettysburg          4
Saga                4
Chaos               4
Cosmic Encounter    4
Gangster            4
Maya                3
Kung Fu             3
Polarity            3
War of the Ring     3
Name: Name, dtype: int64

In [48]:
df_bg[df_bg['Name'] == 'Robin Hood']


ID        Name  Year Published  Min Players  Max Players  \
5352   104640.0  Robin Hood          2011.0            3            5   
11461  258137.0  Robin Hood          2019.0            2            2   
15626    3569.0  Robin Hood          1990.0            2            6   
16474     136.0  Robin Hood          1999.0            3            6   
16545    1947.0  Robin Hood          1991.0            2            6   
19470   31794.0  Robin Hood          1994.0            2            2   

       Play Time  Min Age  Users Rated  Rating Average  BGG Rank  \
5352          50       14          199            7.04      5354   
11461        120       12           39            6.92     11463   
15626         60        8           81            5.66     15628   
16474         30        8           84            5.53     16476   
16545         60       12           61            5.59     16547   
19470         60        0           72            4.63     19472   

       Complexity Average  Owned Users  \
5352                 1.50        262.0   
11461                0.02        135.0   
15626                1.40        207.0   
16474                1.75        221.0   
16545                2.40        190.0   
19470                1.60        109.0   

                                               Mechanics Domains  
5352                       Hand Management, Role Playing     NaN  
11461  Area Majority / Influence, Area Movement, Dice...     NaN  
15626                    Memory, Point to Point Movement     NaN  
16474                                                NaN     NaN  
16545  Action Points, Campaign / Battle Card Driven, ...     NaN  
19470                                       Hexagon Grid     NaN

Okay, turning back to the null ID numbers then.  A quick search confirms these ID numbers are assigned by [boardgamegeek.com](https://boardgamegeek.com).  See this link for for the [Gloomhaven](https://boardgamegeek.com/boardgame/174430/gloomhaven) example. 

In [59]:
df_bg[df_bg['Name'] == 'Gloomhaven'][['ID', 'Name']]


ID        Name
0  174430.0  Gloomhaven

In [311]:
df_null_ids = df_bg[df_bg['ID'].isnull()][['ID', 'Name']].copy()
df_null_ids


ID                                               Name
10776 NaN                            Ace of Aces: Jet Eagles
10835 NaN                                 Die Erben von Hoax
11152 NaN  Rommel in North Africa: The War in the Desert ...
11669 NaN                  Migration: A Story of Generations
12649 NaN                   Die Insel der steinernen Wachter
12764 NaN                   Dragon Ball Z TCG (2014 edition)
13282 NaN                                           Dwarfest
13984 NaN                                                Hus
14053 NaN                                        Contrario 2
14663 NaN                           Warage: Extended Edition
16292 NaN                                            Rainbow
17009 NaN                Sexy, el juego del arte del flirteo
18672 NaN                    Dracarys Dice Don't Get Burned!
19332 NaN  Battleship: Tactical Capital Ship Combat 1925-...
19474 NaN                          The Umbrella Academy Game
20040 NaN                                    Hidden Conflict

In [312]:
df_null_ids["Name"] = df_null_ids["Name"].apply(lambda x: x.replace("Die Insel der steinernen Wachter", "Die Insel der steinernen Wächter"))
df_null_ids["Name"] = df_null_ids["Name"].apply(lambda x: x.replace("Dracarys Dice Don't Get Burned!", "Dracarys Dice"))
df_null_ids


ID                                               Name
10776 NaN                            Ace of Aces: Jet Eagles
10835 NaN                                 Die Erben von Hoax
11152 NaN  Rommel in North Africa: The War in the Desert ...
11669 NaN                  Migration: A Story of Generations
12649 NaN                   Die Insel der steinernen Wächter
12764 NaN                   Dragon Ball Z TCG (2014 edition)
13282 NaN                                           Dwarfest
13984 NaN                                                Hus
14053 NaN                                        Contrario 2
14663 NaN                           Warage: Extended Edition
16292 NaN                                            Rainbow
17009 NaN                Sexy, el juego del arte del flirteo
18672 NaN                                      Dracarys Dice
19332 NaN  Battleship: Tactical Capital Ship Combat 1925-...
19474 NaN                          The Umbrella Academy Game
20040 NaN                                    Hidden Conflict

searching the first item on the list ["Ace of Aces: Jet Eagles"](https://boardgamegeek.com/boardgame/1991/ace-aces-jet-eagles) does list a BGG ID number as 1991.  Is that ID assigned to anything else?

In [ ]:
df_bg[df_bg['ID'] == 1991.0]


Empty DataFrame
Columns: [ID, Name, Year Published, Min Players, Max Players, Play Time, Min Age, Users Rated, Rating Average, BGG Rank, Complexity Average, Owned Users, Mechanics, Domains]
Index: []

In [ ]:
from bs4 import BeautifulSoup
import requests


In [291]:
import requests
from bs4 import BeautifulSoup

title = "Dracarys Dice"

# Function to get the ID for a given title
base_url = "https://boardgamegeek.com/geeksearch.php"
params = {
    "action": "search",
    "advsearch": "1",
    "objecttype": "boardgame",
    "q": title,
}

response = requests.get(base_url, params=params)
soup = BeautifulSoup(response.text, "html.parser")

# Find the link to the board game's page
result = soup.find_all("a", string=title)



In [293]:
result


[<a class="primary" href="/boardgame/269573/dracarys-dice">Dracarys Dice</a>]

In [302]:
# soup


In [300]:
# Function to get the ID for a given title
def get_id_for_title(title):
    base_url = "https://boardgamegeek.com/geeksearch.php"
    params = {
        "action": "search",
        "advsearch": "1",
        "objecttype": "boardgame",
        "q": title,
    }

    response = requests.get(base_url, params=params)
    soup = BeautifulSoup(response.text, "html.parser")
    id_string = soup.find("a", string=title).get("href")
    # return id_string

    if id_string and "/boardgame/" in id_string:
        # Extract the game ID from the href attribute
        game_id = id_string.split("/boardgame/")[1].split("/")[0]
        return int(game_id)
    return None


In [301]:
for i in df_null_ids['Name']:
  print(i)
  print(get_id_for_title(i))
  print( )


Ace of Aces: Jet Eagles
1991

Die Erben von Hoax
413

Rommel in North Africa: The War in the Desert 1941-42
11113

Migration: A Story of Generations
143663

Die Insel der steinernen Wächter
54501

Dragon Ball Z TCG (2014 edition)
168077

Dwarfest
170337

Hus
25999

Contrario 2
27227

Warage: Extended Edition
198886

Rainbow
341510

Sexy, el juego del arte del flirteo
148211

Dracarys Dice
269573

Battleship: Tactical Capital Ship Combat 1925-1945
8173

The Umbrella Academy Game
316555

Hidden Conflict
15804



In [313]:
# Iterate through the DataFrame and update missing IDs
for index, row in df_null_ids.iterrows():
    if pd.isna(row['ID']):
        title = row['Name']
        game_id = get_id_for_title(title)
        df_null_ids.at[index, 'ID'] = game_id

# Display the updated DataFrame
df_null_ids


ID                                               Name
10776    1991.0                            Ace of Aces: Jet Eagles
10835     413.0                                 Die Erben von Hoax
11152   11113.0  Rommel in North Africa: The War in the Desert ...
11669  143663.0                  Migration: A Story of Generations
12649   54501.0                   Die Insel der steinernen Wächter
12764  168077.0                   Dragon Ball Z TCG (2014 edition)
13282  170337.0                                           Dwarfest
13984   25999.0                                                Hus
14053   27227.0                                        Contrario 2
14663  198886.0                           Warage: Extended Edition
16292  341510.0                                            Rainbow
17009  148211.0                Sexy, el juego del arte del flirteo
18672  269573.0                                      Dracarys Dice
19332    8173.0  Battleship: Tactical Capital Ship Combat 1925-...
19474  316555.0                          The Umbrella Academy Game
20040   15804.0                                    Hidden Conflict

In [315]:
df_bg["Name"] = df_bg["Name"].apply(lambda x: x.replace("Die Insel der steinernen Wachter", "Die Insel der steinernen Wächter"))
df_bg["Name"] = df_bg["Name"].apply(lambda x: x.replace("Dracarys Dice Don't Get Burned!", "Dracarys Dice"))

# Iterate through the DataFrame and update missing IDs
for index, row in df_bg.iterrows():
    if pd.isna(row['ID']):
        title = row['Name']
        game_id = get_id_for_title(title)
        df_bg.at[index, 'ID'] = game_id

# Display the updated DataFrame
df_bg.isnull().sum()


ID                        0
Name                      0
Year Published            1
Min Players               0
Max Players               0
Play Time                 0
Min Age                   0
Users Rated               0
Rating Average            0
BGG Rank                  0
Complexity Average        0
Owned Users              23
Mechanics              1598
Domains               10159
dtype: int64

In [ ]:
.replace(' ', '%20')


In [ ]:
# Iterate through the DataFrame and update missing IDs
for index, row in df_null_ids.iterrows():
    if pd.isna(row['ID']):
        title = row['Name']
        game_id = get_id_for_title(title)
        df_null_ids.at[index, 'ID'] = game_id

# Display the updated DataFrame
print(df_null_ids)


IndexError: list index out of range

In [ ]:

# Create a DataFrame with the data you provided
data = {
    'ID': [None] * 15,
    'Name': [
        "Ace of Aces: Jet Eagles",
        "Die Erben von Hoax",
        "Rommel in North Africa: The War in the Desert ...",
        "Migration: A Story of Generations",
        "Die Insel der steinernen Wachter",
        "Dragon Ball Z TCG (2014 edition)",
        "Dwarfest",
        "Hus",
        "Contrario 2",
        "Warage: Extended Edition",
        "Rainbow",
        "Sexy, el juego del arte del flirteo",
        "Dracarys Dice Don't Get Burned!",
        "Battleship: Tactical Capital Ship Combat 1925-...",
        "The Umbrella Academy Game",
        "Hidden Conflict"
    ]
}

df_null_ids = pd.DataFrame(data)



In [ ]:
import pandas as pd
import requests
from bs4 import BeautifulSoup


In [ ]:
df_null_ids['Name'].tolist()


['Ace of Aces: Jet Eagles',
 'Die Erben von Hoax',
 'Rommel in North Africa: The War in the Desert 1941-42',
 'Migration: A Story of Generations',
 'Die Insel der steinernen Wachter',
 'Dragon Ball Z TCG (2014 edition)',
 'Dwarfest',
 'Hus',
 'Contrario 2',
 'Warage: Extended Edition',
 'Rainbow',
 'Sexy, el juego del arte del flirteo',
 "Dracarys Dice Don't Get Burned!",
 'Battleship: Tactical Capital Ship Combat 1925-1945',
 'The Umbrella Academy Game',
 'Hidden Conflict']

In [ ]:
import pandas as pd
import requests
from bs4 import BeautifulSoup

# Your DataFrame
data = {
    'ID': [None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None]
    'Name': [
        'Ace of Aces: Jet Eagles',
        'Die Erben von Hoax',
        'Rommel in North Africa: The War in the Desert 1941-42',
        'Migration: A Story of Generations',
        'Die Insel der steinernen Wachter',
        'Dragon Ball Z TCG (2014 edition)',
        'Dwarfest',
        'Hus',
        'Contrario 2',
        'Warage: Extended Edition',
        'Rainbow',
        'Sexy, el juego del arte del flirteo',
        "Dracarys Dice Don't Get Burned!",
        'Battleship: Tactical Capital Ship Combat 1925-1945',
        'The Umbrella Academy Game',
        'Hidden Conflict'
    ]
}

df = pd.DataFrame(data)

# Function to get the ID for a given title
def get_id_for_title(title):
    base_url = "https://boardgamegeek.com/geeksearch.php"
    params = {
        "action": "search",
        "advsearch": "1",
        "objecttype": "boardgame",
        "q": title,
    }

    response = requests.get(base_url, params=params)
    soup = BeautifulSoup(response.text, "html.parser")
    id_link = soup.find("a", href=True)

    if id_link:
        return int(id_link["href"].split('/')[-1])
    else:
        return None

# Iterate through the DataFrame and update missing IDs
for index, row in df_null_ids.iterrows():
    if pd.isna(row['ID']):
        title = row['Name']
        game_id = get_id_for_title(title)
        df_null_ids.at[index, 'ID'] = game_id

# Display the updated DataFrame
print(df_null_ids)


SyntaxError: invalid syntax. Perhaps you forgot a comma? (1673805328.py, line 7)

In [ ]:
# Function to get the ID for a given title
def get_id_for_title(title):
    base_url = "https://boardgamegeek.com/geeksearch.php"
    params = {
        "action": "search",
        "advsearch": "1",
        "objecttype": "boardgame",
        "q": title,
    }

    response = requests.get(base_url, params=params)
    soup = BeautifulSoup(response.text, "html.parser")
    id_link = soup.find("a", href=True)

    if id_link:
        return int(id_link["href"].split('/')[-1])
    else:
        return None


In [ ]:
df_null_ids['Name'].iloc[0].replace(' ', '%20')


'Ace%20of%20Aces:%20Jet%20Eagles'

In [ ]:
# Iterate through the DataFrame and update missing IDs
for index, row in df_null_ids.iterrows():
    if pd.isna(row['ID']):
        title = row['Name'].replace(' ', '%20')
        game_id = get_id_for_title(title)
        df_null_ids.at[index, 'ID'] = game_id

# Display the updated DataFrame
print(df_null_ids)


ValueError: invalid literal for int() with base 10: 'page'

In [ ]:
# Function to get the ID for a given title
def get_id_for_title(title):
    base_url = "https://boardgamegeek.com/geeksearch.php"
    params = {
        "action": "search",
        "advsearch": "1",
        "objecttype": "boardgame",
        "q": title,
    }

    response = requests.get(base_url, params=params)
    soup = BeautifulSoup(response.text, "html.parser")
    id_link = soup.find("a", href=True)

    if id_link:
        game_url = id_link["href"]
        if "/boardgame/" in game_url:
            game_id = int(game_url.split('/')[-2])
            return game_id
    return None

# Iterate through the DataFrame and update missing IDs
for index, row in df_null_ids.iterrows():
    if pd.isna(row['ID']):
        title = row['Name'].replace(' ', '%20')
        print(title)
        game_id = get_id_for_title(title)
        print(game_id)
        df_null_ids.at[index, 'ID'] = game_id

# Display the updated DataFrame
print(df_null_ids)


Ace%20of%20Aces:%20Jet%20Eagles
None
Die%20Erben%20von%20Hoax
None
Rommel%20in%20North%20Africa:%20The%20War%20in%20the%20Desert%201941-42
None
Migration:%20A%20Story%20of%20Generations
None
Die%20Insel%20der%20steinernen%20Wachter
None
Dragon%20Ball%20Z%20TCG%20(2014%20edition)
None
Dwarfest
None
Hus
None
Contrario%202
None
Warage:%20Extended%20Edition
None
Rainbow


ValueError: invalid literal for int() with base 10: 'page'

In [ ]:
https://boardgamegeek.com/geeksearch.php?action=search&advsearch=1&objecttype=boardgame&q=Ace%20of%20Aces:%20Jet%20Eagles&include%5Bdesignerid%5D=&geekitemname=&geekitemname=&include%5Bpublisherid%5D=&range%5Byearpublished%5D%5Bmin%5D=&range%5Byearpublished%5D%5Bmax%5D=&range%5Bminage%5D%5Bmax%5D=&floatrange%5Bavgrating%5D%5Bmin%5D=&floatrange%5Bavgrating%5D%5Bmax%5D=&range%5Bnumvoters%5D%5Bmin%5D=&floatrange%5Bavgweight%5D%5Bmin%5D=&floatrange%5Bavgweight%5D%5Bmax%5D=&range%5Bnumweights%5D%5Bmin%5D=&colfiltertype=&searchuser=&range%5Bminplayers%5D%5Bmax%5D=&range%5Bmaxplayers%5D%5Bmin%5D=&playerrangetype=normal&range%5Bleastplaytime%5D%5Bmin%5D=&range%5Bplaytime%5D%5Bmax%5D=&B1=Submit


In [ ]:
<!DOCTYPE html>
<html ng-app="GeekApp" lang="en-US" ng-cloak>

<head>
	<meta charset='utf-8'>
	<meta id="vp" name="viewport" content="width=device-width, initial-scale=1.0">
	<script>
		window.addEventListener( 'DOMContentLoaded',  function() {
				var width = document.documentElement.clientWidth || window.innerWidth;
				if (width < 960) {
					var mvp = document.getElementById('vp');
					// android debugging
					mvp.setAttribute('content','width=960');
				}
			});
	</script>
	<meta content='yes' name='apple-mobile-web-app-capable'>
	<meta content='IE=edge,chrome=1' http-equiv='X-UA-Compatible'>

	<title>BoardGameGeek | Gaming Unplugged Since 2000</title>


	<link rel="apple-touch-icon" href="https://cf.geekdo-static.com/icons/touch-icon180.png" />
	<link rel="shortcut icon" href="https://cf.geekdo-static.com/icons/favicon2.ico" type="image/ico" />
	<link rel="icon" href="https://cf.geekdo-static.com/icons/favicon2.ico" type="image/ico" />
	<link rel="search" href="/game-opensearch.xml" type="application/opensearchdescription+xml"
		title="BGG Game Search" />
	<meta name="apple-mobile-web-app-title" content="BGG">

	<meta name="theme-color" content="#2e2b47">
	<link rel="preconnect" href="https://api.geekdo.com" />




	<meta property="og:image" content="https://cf.geekdo-static.com/images/opengraph/bgg_opengraph.png" />



	<meta name="keywords"
		content="board game, boardgames, boardgame, board, games, game, hobby, boardgamegeek, geek, geekdo">

	<script>
		window.AdSlots = window.AdSlots || {
	cmd: [],
	disableScripts: ['gpt'],
	renderOnFirstLoad: false,
	divCheck: false
};
	</script>


	<link rel='stylesheet' type='text/css' href='https://cf.geekdo-static.com/static/geekui_master2_6526f5bac1c35.css'>
	<link rel='stylesheet' type='text/css' href='https://cf.geekdo-static.com/static/css_master2_6526f5ad38a8d.css'>

	<base href="/">

	<script src="https://kit.fontawesome.com/ece5b35462.js" crossorigin="anonymous"></script>

	<script>
		var GEEK = GEEK || {};
	GEEK.adBlock = [];
	GEEK.adConfig = {"blockleaderboard":null,"blockskyscraper":null,"noadsense":null,"blocksupportdrive":null};
	GEEK.adSlots = {"dfp-leaderboard":{"name":"boardgame_leaderboard_728x90"},"dfp-skyscraper":{"name":"boardgame_skyscraper_160x600"},"dfp-medrect":{"name":"boardgame_rectangle_300x250"},"dfp-repeater":{"name":"boardgame_home_repeater"},"dfp-medrect-reserved-home":{"name":"boardgame_reserved_home_300x250"},"dfp-leaderboard-lg":{"name":"boardgame_home_hero"},"dfp-home-sidekick":{"name":"boardgame_home_repeater"},"dfp-inline-post":{"name":"boardgame_inline_post"}};
	GEEK.legacyAds = [];
	GEEK.bggStoreAds = [];
	GEEK.googleTargets = null;
	GEEK.userid = 0;
	GEEK.domainname = 'boardgamegeek.com';
	GEEK.domain = 'boardgame';
	GEEK.geekitemPreload = {};
	GEEK.geekitemSettings = null;
	GEEK.geekitemModules = null;
	GEEK.geekGlobalSettings = {"shutdown_file_ops":"1","shutdown_storeimage_ops":"0","shutdown_edit_avatar":"0","shutdown_file_upload":"0","shutdown_file_download":"0","shutdown_image_upload":"0"};
	GEEK.geekimageSettings = null;
	GEEK.legacy = 1;
	GEEK.apiurlsPrefix = 'https://api.geekdo.com';



		GEEK.apiurls = {
		'root': '/api',
		'amazon': '/api/amazon',
		'files': '/api/files',
		'geekitems': '/api/geekitems',
		'images': '/api/images',
		'threads': '/api/forums/threads',
	 	'forums': '/api/forums',
		'videos': '/api/videos',
		'hotness': '/api/hotness',
		'dynamicinfo': '/api/dynamicinfo',
		'subtypeinfo': '/api/subtypeinfo',
		'geekbay': '/api/geekbay',
		'geekmarket': '/market/api/v1',
	    'geekmarketapi': '/api/market',
		'geeklists': '/api/geeklists',
		'reviews': '/api/forumreviews',
		'collections': '/api/collections',
		'linkeditems': '/api/geekitem/linkeditems',
		'subscriptions': '/api/subscriptions',
	    'fans': '/api/fans',
	    'geekpreviews': '/api/geekpreviews',
	    'geekpreviewitems': '/api/geekpreviewitems',
	   	'geekpreviewparentitems': '/api/geekpreviewparentitems',
		'recs': '/api/geekitem/recs',
	    'awards': '/api/geekawards',
		'historicalrankgraph':  '/api/historicalrankgraph',
		'blueprint_recipes': '/api/blueprints/recipes',
		'affiliateads' : '/api/affiliateads',
		'sleevesbycard': '/api/sleevesbycard',
		'cardsetsbygame': '/api/cardsetsbygame',
	};
	</script>

	<script async src="https://www.googletagmanager.com/gtag/js?id=UA-104725-1"></script>
	<script>
		window.dataLayer = window.dataLayer || [];
	function gtag(){dataLayer.push(arguments);}
	gtag('js', new Date());
	gtag('consent', 'default', {
		'ad_storage': 'denied',
		'analytics_storage': 'denied',
		'personalization_storage': 'denied',
		'wait_for_update': 3000
		 });
	gtag('config', 'UA-104725-1', {
		'cookie_domain': 'boardgamegeek.com',
		'send_page_view': false
	});
	</script>

	<script type='text/javascript' src='https://cf.geekdo-static.com/static/geekangular17_master2_6526f5c7e437f.js'>
	</script>
	<script type='text/javascript' src='https://cf.geekdo-static.com/static/geekui_master2_6526f5bac1c35.js'></script>
	<script type='text/javascript' src='https://cf.geekdo-static.com/static/js_master2_6526f5acc8197.js'></script>
	<script type='text/javascript'
		src='https://cf.geekdo-static.com/static/geekoutputtemplates_master2_6526f5c89c062.js'></script>
	<script type='text/javascript'
		src='https://cf.geekdo-static.com/static/geekuicommontemplates_master2_6526f5c8a49e3.js'></script>
	<script>
		window.geekCookieConsent.gtagReady.then(function() {
		gtag('event', 'page_view');
	});
	</script>
	<!-- why this? -->
	<meta http-equiv="content-type" content="text/html;charset=UTF-8">

	<!-- Add to main?
<link rel="apple-touch-icon" 	href="https://cf.geekdo-static.com/icons/appleicon.png" />
<link rel="shortcut icon" 		href="https://cf.geekdo-static.com/icons/favicon.ico" type="image/ico" />
<link rel="icon" 					href="https://cf.geekdo-static.com/icons/favicon.ico" type="image/ico" />
<link rel="search" 				href="/game-opensearch.xml" type="application/opensearchdescription+xml" title="BGG Game Search" />
-->





	<!--this is used only for GeekMap -->


	<!-- deal with adsense
	<script type="text/javascript">
		window.google_analytics_uacct = "UA-104725-1";
		var googletag = googletag || { };
		googletag.cmd = googletag.cmd || [];
		( function() {
			var gads = document.createElement('script');
			gads.async = true;
			gads.type  = 'text/javascript';
			gads.src   = "//www.googletagservices.com/tag/js/gpt.js";
			var node = document.getElementsByTagName('script')[0];
			node.parentNode.insertBefore(gads, node);
		} )();

		adunits = [
			{
				name:    'boardgame_button_120x45',
				size:    [ 120, 45 ],
				target:  'dfp-button'
			},
			{
				name:    'boardgame_leaderboard_728x90',
				size:    [ 728, 90 ],
				target:  'dfp-leaderboard'
			},
			{
				name:    'boardgame_skyscraper_160x600',
				size:    [ 160, 600 ],
				target:  'dfp-skyscraper'
			},
			{
				name:    'boardgame_rectangle_300x250',
				size:    [ 300, 250 ],
				target:  'dfp-medrect'
			},
			{
				name:    'boardgame_rectangle_300x250',
				size:    [ 300, 250 ],
				target:  'dfp-medrect-reserved-home'
			},
			{
				name:    'boardgame_button_120x90',
				size:    [ 120, 90 ],
				target:  'dfp-giftguide'
			}
		];

		googletag.cmd.push(function() {
			for( var i=0; i< adunits.length; i++ )
			{
				unit = adunits[i];
				googletag.defineUnit('/1005854/ca-pub-7206761047394129/'+unit.name, unit.size, unit.target).
				addService(googletag.pubads());
			}


			googletag.pubads().setTargeting( "subdomain", "all" );
			googletag.enableServices();
		} );

	</script>
-->

	<script>
		function start() {

	}

	function ondomready() {
				Amazon_LoadAds();

		GEEK.addHandlers();
		GEEK.recaptchaKey = '6Ldyr6EaAAAAAE0F2hYgtHHqbF6nKPTENAwo6SyU';
	}

	if( typeof window.addEvent === "function" ) {
		window.addEvent('domready', function() {
			ondomready();
		} );
	} else {
		window.addEventListener( "DOMContentLoaded", function() {
			ondomready();
	} );
	}
	window.onload = start;
	</script>


</head>

<body ng-controller="GeekOutput_LayoutCtrl as layoutctrl" class="domain-boardgame"
	ng-class="{ 'has-no-max-width' : layoutctrl.geekitemSettings.fluidlayout }">


	<div class='d-flex flex-column min-vh-100'>

		<div id="global-header-outer" class='global-header-outer' ng-controller="NavCtrl as navctrl"
			click-out="navctrl.closeMobileMenu()">

			<geeknav-menu></geeknav-menu>
		</div>



		<main class='global-body flex-grow-1' id="mainbody"
			ng-class="{ 'has-overlay-sidebar': layoutctrl.showOverlaySidebar, 'has-hidden-fixed-sidebar': layoutctrl.localStorage.hideFixedSidebar }">

			<a id='mainbodytarget' tabindex="-1"></a>

			<!-- Home King Ad -->

			<div hide-ad-block="blockleaderboard">
				<div class="advertisement-leaderboard">
					<div class='center-block' ng-dfp-ad="dfp-leaderboard"></div>
				</div>
			</div>

			<div class='global-body-content-container container-fluid'>
				<geekoutput-sidebar deactivate-overlay-sidebar='layoutctrl.deactivateOverlaySidebar'
					show-overlay-sidebar='layoutctrl.showOverlaySidebar'></geekoutput-sidebar>


				<div class='global-body-content pending' ng-class="{'ready': layoutctrl.ready}">

					<a id='maincontenttarget' tabindex="-1"></a>

					<div class="legacy">
						<div id="container" class="yui-skin-sam">
							<div id="maincontent" ng-non-bindable>
								<div style="margin: auto;"></div>
								<form method='GET' action="">
									<div class='infobox'>
										<div class='fr'></div>
										<div class='fl'>
											<p class="m-0 text-muted">We may earn a commission when you buy through our
												links. </p>
										</div>

										<div class='clear'></div>
									</div>
								</form>

								<div id='collection_status' style='position:absolute; background:red; color:white;'>
								</div>


								<div id='collection'>
									<div class='fl sf'>
										<span id='collection_viewlabel'style='display:none;'>&nbsp;|&nbsp;View: <span id='collection_viewname'></span></span>
									</div>

									<div style='text-align:right; margin-bottom:5px;' class='sf'>
									</div>



									<div class="table-responsive">
										<table cellpadding=0 cellspacing=1 class='collection_table' width='100%'
											id='collectionitems'>
											<tr>
												<th class='collection_bggrating'>

													<a
														href="/search/boardgame?sort=rank&advsearch=1&q=Ace+of+Aces%3A+Jet+Eagles&include%5Bdesignerid%5D=&include%5Bpublisherid%5D=&geekitemname=&range%5Byearpublished%5D%5Bmin%5D=&range%5Byearpublished%5D%5Bmax%5D=&range%5Bminage%5D%5Bmax%5D=&range%5Bnumvoters%5D%5Bmin%5D=&range%5Bnumweights%5D%5Bmin%5D=&range%5Bminplayers%5D%5Bmax%5D=&range%5Bmaxplayers%5D%5Bmin%5D=&range%5Bleastplaytime%5D%5Bmin%5D=&range%5Bplaytime%5D%5Bmax%5D=&floatrange%5Bavgrating%5D%5Bmin%5D=&floatrange%5Bavgrating%5D%5Bmax%5D=&floatrange%5Bavgweight%5D%5Bmin%5D=&floatrange%5Bavgweight%5D%5Bmax%5D=&colfiltertype=&searchuser=&playerrangetype=normal&B1=Submit">Board
														Game Rank</a>
												</th>

												<th class='collection_thumbnail'>
													<span class="sr-only">Thumbnail image</span>
												</th>

												<th class='collection_title'>

													<a
														href="/search/boardgame?sort=title&advsearch=1&q=Ace+of+Aces%3A+Jet+Eagles&include%5Bdesignerid%5D=&include%5Bpublisherid%5D=&geekitemname=&range%5Byearpublished%5D%5Bmin%5D=&range%5Byearpublished%5D%5Bmax%5D=&range%5Bminage%5D%5Bmax%5D=&range%5Bnumvoters%5D%5Bmin%5D=&range%5Bnumweights%5D%5Bmin%5D=&range%5Bminplayers%5D%5Bmax%5D=&range%5Bmaxplayers%5D%5Bmin%5D=&range%5Bleastplaytime%5D%5Bmin%5D=&range%5Bplaytime%5D%5Bmax%5D=&floatrange%5Bavgrating%5D%5Bmin%5D=&floatrange%5Bavgrating%5D%5Bmax%5D=&floatrange%5Bavgweight%5D%5Bmin%5D=&floatrange%5Bavgweight%5D%5Bmax%5D=&colfiltertype=&searchuser=&playerrangetype=normal&B1=Submit">Title</a>
												</th>



												<th class='collection_bggrating'>
													<a
														href="/search/boardgame?sort=bggrating&advsearch=1&q=Ace+of+Aces%3A+Jet+Eagles&include%5Bdesignerid%5D=&include%5Bpublisherid%5D=&geekitemname=&range%5Byearpublished%5D%5Bmin%5D=&range%5Byearpublished%5D%5Bmax%5D=&range%5Bminage%5D%5Bmax%5D=&range%5Bnumvoters%5D%5Bmin%5D=&range%5Bnumweights%5D%5Bmin%5D=&range%5Bminplayers%5D%5Bmax%5D=&range%5Bmaxplayers%5D%5Bmin%5D=&range%5Bleastplaytime%5D%5Bmin%5D=&range%5Bplaytime%5D%5Bmax%5D=&floatrange%5Bavgrating%5D%5Bmin%5D=&floatrange%5Bavgrating%5D%5Bmax%5D=&floatrange%5Bavgweight%5D%5Bmin%5D=&floatrange%5Bavgweight%5D%5Bmax%5D=&colfiltertype=&searchuser=&playerrangetype=normal&B1=Submit">Geek
														Rating</a>

												</th>

												<th class='collection_bggrating'>
													<a
														href="/search/boardgame?sort=avgrating&advsearch=1&q=Ace+of+Aces%3A+Jet+Eagles&include%5Bdesignerid%5D=&include%5Bpublisherid%5D=&geekitemname=&range%5Byearpublished%5D%5Bmin%5D=&range%5Byearpublished%5D%5Bmax%5D=&range%5Bminage%5D%5Bmax%5D=&range%5Bnumvoters%5D%5Bmin%5D=&range%5Bnumweights%5D%5Bmin%5D=&range%5Bminplayers%5D%5Bmax%5D=&range%5Bmaxplayers%5D%5Bmin%5D=&range%5Bleastplaytime%5D%5Bmin%5D=&range%5Bplaytime%5D%5Bmax%5D=&floatrange%5Bavgrating%5D%5Bmin%5D=&floatrange%5Bavgrating%5D%5Bmax%5D=&floatrange%5Bavgweight%5D%5Bmin%5D=&floatrange%5Bavgweight%5D%5Bmax%5D=&colfiltertype=&searchuser=&playerrangetype=normal&B1=Submit">Avg
														Rating</a>

												</th>

												<th class='collection_bggrating'>
													<a
														href="/search/boardgame?sort=numvoters&advsearch=1&q=Ace+of+Aces%3A+Jet+Eagles&include%5Bdesignerid%5D=&include%5Bpublisherid%5D=&geekitemname=&range%5Byearpublished%5D%5Bmin%5D=&range%5Byearpublished%5D%5Bmax%5D=&range%5Bminage%5D%5Bmax%5D=&range%5Bnumvoters%5D%5Bmin%5D=&range%5Bnumweights%5D%5Bmin%5D=&range%5Bminplayers%5D%5Bmax%5D=&range%5Bmaxplayers%5D%5Bmin%5D=&range%5Bleastplaytime%5D%5Bmin%5D=&range%5Bplaytime%5D%5Bmax%5D=&floatrange%5Bavgrating%5D%5Bmin%5D=&floatrange%5Bavgrating%5D%5Bmax%5D=&floatrange%5Bavgweight%5D%5Bmin%5D=&floatrange%5Bavgweight%5D%5Bmax%5D=&colfiltertype=&searchuser=&playerrangetype=normal&B1=Submit">Num
														Voters</a>

												</th>





















												<th class='collection_shop'>
													Shop
												</th>
											</tr>






											<tr id='row_'>
												<td class='collection_rank' align='center'>
													<a name="12855"></a> 12855
												</td>


												<td class='collection_thumbnail'>
													<a href="/boardgame/1991/ace-aces-jet-eagles"><img alt="Board Game: Ace of Aces: Jet Eagles"   src="https://cf.geekdo-images.com/aPlzfmmKbWpTCqTNZOYqcQ__micro/img/Yzf1Rp9jCCso-Bd0NFidzG7n7OU=/fit-in/64x64/filters:strip_icc()/pic6183029.jpg"></a>
												</td>

												<td id='CEcell_objectname1' class="collection_objectname
						 browse">

													<div id='status_objectname1'></div>
													<div id='results_objectname1' style='z-index:1000;' onclick=''>

														<a href="/boardgame/1991/ace-aces-jet-eagles"
															class='primary'>Ace of Aces: Jet Eagles</a>

														<span class='smallerfont dull'>(1990)</span>

													</div>

													<p class="smallefont dull" style="margin: 2px 0 0 0;">
														Hop into the cockpit of a jet fighter in this variant of the
														innovative Ace of Aces.
													</p>
												</td>



												<td class='collection_bggrating' align='center'>
													5.539 </td>

												<td class='collection_bggrating' align='center'>
													6.34 </td>

												<td class='collection_bggrating' align='center'>
													122 </td>



























												<td class='collection_shop'>
													<div class='aad' id='aad_thing_1991_textwithprices__'></div>
												</td>

											</tr>


										</table>
									</div>

								</div>


								<p align='right'></p>

								<div id="legacy_modal"></div>
							</div>
						</div>
					</div>

					<div class='global-body-content-secondary'>
					</div>
				</div>
			</div>

		</main>
		<geekoutput-footer></geekoutput-footer>
	</div>




</body>

</html>


In [ ]:
df_bg['Mechanics'][0]


'Action Queue, Action Retrieval, Campaign / Battle Card Driven, Card Play Conflict Resolution, Communication Limits, Cooperative Game, Deck Construction, Deck Bag and Pool Building, Grid Movement, Hand Management, Hexagon Grid, Legacy Game, Modular Board, Once-Per-Game Abilities, Scenario / Mission / Campaign Game, Simultaneous Action Selection, Solo / Solitaire Game, Storytelling, Variable Player Powers'

## 💪 Challenge
Explore and analyze the board game data, and share the intriguing insights with your friends through a report. Here are some steps that might help you get started:

* Is this dataset ready for analysis? Some variables have inappropriate data types, and there are outliers and missing values. Apply data cleaning techniques to preprocess the dataset.
* Use data visualization techniques to draw further insights from the dataset. 
* Find out if the number of players impacts the game's average rating.

## 🧑‍⚖️ Judging criteria

This is a community-based competition. The top 5 most upvoted entries will win.

The winners will receive DataCamp merchandise.

## ✅ Checklist before publishing into the competition
- Rename your workspace to make it descriptive of your work. N.B. you should leave the notebook name as notebook.ipynb.
- **Remove redundant cells** like the judging criteria, so the workbook is focused on your story.
- Make sure the workbook reads well and explains how you found your insights. 
- Try to include an **executive summary** of your recommendations at the beginning.
- Check that all the cells run without error.

## ⌛️ Time is ticking. Good luck!